### ISSUE TITLE
Edit fails to add element to array if it's previously empty

### ISSUE THREAD

**Issue by jimmidyson** [fabric8io/kubernetes-client#502]
From @josefkarasek's comment in #491 https://github.com/fabric8io/kubernetes-client/issues/491#issuecomment-243041311:

> Failing to add to users when the list is empty. Probably affects all other lists.
> The following operation fails: https://github.com/kubernetes/kubernetes/blob/59f918cae0105a55efa515d0cd7dfe3b56ead627/vendor/github.com/evanphx/json-patch/patch.go#L415

I think this is correct behaviour as per the JSON Patch spec. Think we need to send through the add with the new single element array if the array doesn't prevously exist. This should be done in our zjsonpatch fork at https://github.com/fabric8io/zjsonpatch.

**Comment by jimmidyson** [comment]
Hmm zjsonpatch also works properly... the problem is that in our generated kubernetes-model we initialize all collections so even if the API server sends back a null (nil, whatever you want to call it) value for a collection, it is initialized as an empty collection, not null.

Thoughts @iocanel? I'd prefer the model to accurately represent what the API server has returned, otherwise we can't calculate patches correctly.

**Comment by lwander** [comment]
Hey, I've run into this same issue: 

```
Failure executing: PATCH at: https://<endpoint>/apis/extensions/v1beta1/namespaces/default/deployments/hook. Message: jsonpatch add operation does not apply: doc is missing path: /spec/template/spec/containers/0/command/0. Received status: Status(apiVersion=v1, code=500, details=null, kind=Status, message=jsonpatch add operation does not apply: doc is missing path: /spec/template/spec/containers/0/command/0, metadata=ListMeta(resourceVersion=null, selfLink=null, additionalProperties={}), reason=null, status=Failure, additionalProperties={}).
```

**Comment by jimmidyson** [comment]
Yeah we haven't fixed this up yet. @iocanel What do you think about my suggestion above about null vs empty collections?

**Comment by Fsero** [comment]
I've also run into the same issue.

**Comment by osoriano** [comment]
Having initializeCollections=true also causes other issues such as https://github.com/fabric8io/kubernetes-client/issues/931

**Comment by jglick** [comment]
Is this issue closed as of #615?
